src/test/java/codeu/model/data/ModelDataTestHelpers.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.model.data;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class ModelDataTestHelpers {
+
+  /** Asserts that all fields on both Messages are the same. */
+  public static void assertMessageEquals(Message expected, Message actual) {
```

Comment:
Another option would be to add an `equals()` method to the `Message` class. I originally avoided that because I didn't want to also implement `hashCode()` and then never use it, but I'm not so sure that was the way to go. What do you think?

Reply:
Eclipse can autogenerate both equals and hashCode.  The only issue with them is that you have to remember to update them if you ever add any fields.  They're not hard to update, but if you don't do it then you'll end up with surprising results.